pipelines/oracle_grounded/oracles.py

```
@@ -0,0 +1,701 @@
+    return returncode, stdout
+
+
+class ChainOracle(OracleAdapter):
+    """An ordered pipeline of oracles, e.g. limbic-critic -> plasticity-lab.
+
+    Each step is resolved independently, so a deployment that has one of the
+    two runtimes bound gets a record whose ``oracle.stages`` says exactly which
+    half was measured by the named runtime and which half was the reference.
+    """
+
+    def __init__(self, oracle_id, oracle_type, description, steps, version="1.0.0"):
```

Ruff (0.16.2)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)